### DISCUSSION TITLE
In i18nextify, how do I get back the detected/previously stored language?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [i18next/i18next#2073] [Q&A]
Hi, I am new to i18next/i18nextify. 

1. I would like to load the translations from the local storage first with fall-back to http backend resource - is this the correct way to do it:

```javascript
const langVersions = { 
  en: 'v1.0', 
  bg: 'v1.0', 
  ru: 'v1.0' 
};

window.i18nextify.init({
  fallbackLng: "en",
    backend: {
      backends: [
        LocalStorageBackend,
        HttpApi
      ],
      backendOptions: [{
        //LocalStorageBackend config
        expirationTime: 100 * 24 * 60 * 60 * 1000, // 100 days
        versions: langVersions
      }, {
        //HttpApi config
        loadPath: '/locales/{{lng}}/{{ns}}.json',
        versions: langVersions
      }]
    }
});
```

2. I need to get the active language (whether detected, or stored from before) in order to show it on the UI - how do I do that. AI tools tell me this - is this the correct way:

```javascript
window.i18nextify.init({ ...});
const languageCode = window.i18nextify.getLanguage();
// set the value in my UI to display to the user
```

**Comment by jamuhl** [comment]
Looks ok to me...if it works it is correct - if not...idk...hard to run the code based on a snippet.

Regarding accessing the current lng: https://github.com/i18next/i18nextify/tree/master#translating-javascript-code

You can access `window.i18nextify.i18next` which is the i18next instance used - on that you can use stuff from https://www.i18next.com/overview/api --> https://www.i18next.com/overview/api#resolvedlanguage might be what you search

**Comment by hrstoyanov** [comment]
Should there be a post-init call back, something like:
```javascript
window
  .i18nextify
  .init({
      debug: true,
      fallbackLng: 'en',
      backend: {
        loadPath: '/locales/{{lng}}/{{ns}}.json'
      }
  }).then(...)
```

I see nothing in the docs?

Without it, I am getting this:
```
i18next::translator: key "TourBiz logo icon" for languages "bg, en" won't get resolved as namespace "translation" was not yet loaded This means something IS WRONG in your setup. You access the t function before i18next.init / i18next.loadNamespace / i18next.changeLanguage was done. Wait for the callback or Promise to resolve before accessing it!!!
```

**Reply by hrstoyanov** [reply]
Ok ..fixed by ordering my <script> tags
